### PR TITLE
fix Long.signum return type

### LINKED ISSUE #1550: Long.signum has wrong return type
Long.signum has wrong return type - currently it is ```scala.Long``` but should be ```Int```

I have a PR for this coming very soon...

### PR DESCRIPTION
**Pull request by InTheNow**
fixes #1550

### CODE REVIEW

**Comment by sjrd** [comment]
Thanks.

This needs a test in https://github.com/scala-js/scala-js/blob/master/test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala

The commit message should be:
```
Fix #1550: Make Long.signum return an Int.
```

And finally, it'll have to wait until the "Towards 0.6.3." commit lands on master, and be rebased on top of that one.

**Comment by sjrd** [comment]
> And finally, it'll have to wait until the "Towards 0.6.3." commit lands on master, and be rebased on top of that one.

It has landed. To rebase:
```sh
$ git fetch scalajs # or the name of your remote for scala-js/scala-js
$ git rebase scalajs/master
```

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala [hunk 1]
```diff
@@ -174,5 +174,30 @@ object LongTest extends JasmineTest {
...
+      expect(JLong.signum(-98765432158845L)).toEqual(-1)
+      expect(JLong.signum(-49575304457780L)).toEqual(-1)
+      expect(JLong.signum(-11L)).toEqual(-1)
+      expect(JLong.signum(-1L)).toEqual(-1)
+      expect(JLong.signum(0L)).toEqual(0)
+      expect(JLong.signum(1L)).toEqual(1)
+      expect(JLong.signum(11L)).toEqual(1)
+      expect(JLong.signum(49575304457780L)).toEqual(1)
+      expect(JLong.signum(98765432158845L)).toEqual(1)
+      expect(JLong.signum(Long.MaxValue)).toEqual(1)
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala]: OK, that's more than I expected ^^ But all the better!

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala [hunk 2]
```diff
@@ -174,5 +174,30 @@ object LongTest extends JasmineTest {
...
+
+      // check return type
+      expect(JLong.signum(1).isInstanceOf[Int]).toBeTruthy
+      expect(JLong.signum(1).isInstanceOf[Long]).toBeFalsy
```
Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/LongTest.scala]: These two tests should not be there, though. This is the job of the typechecker+linker. If the test suite links, then it is guaranteed to be true. So no runtime test for this.

**Comment by sjrd** [comment]
The two (or more) commits should also be squashed together, keeping the commit message I suggested.

**Comment by InTheNow** [comment]
done

**Comment by sjrd** [comment]
LGTM thanks.

**Comment by sjrd** [comment]
retest this please

**Pull request merged**